### PR TITLE
Allow host to be specified for OAuth 2 token refresh

### PR DESCRIPTION
**Pull request by tjwp**
Most of the changes in `refresh_spec.rb` are just moving existing specs into a describe block.

### CODE REVIEW

File: spec/lib/refresh_spec.rb [hunk 1]
```diff
@@ -26,57 +26,93 @@
...
     end
 
-    it 'calls the callback with the response body' do
-      connection.stub(:post).and_return(success_response)
-      callback.should_receive(:call).with('access_token' => 'refreshed')
+    context 'when an authentication callback is specified' do
+      let(:callback) { double('Proc') }
 
-      trigger
+      before(:each) do
```
Comment (hankwallace) [inline spec/lib/refresh_spec.rb:97]: Is there a preference for specifying `(:each)` or leaving it out because it's the default?

**Comment by hankwallace** [comment]
Just one question. Looks good.

File: spec/lib/refresh_spec.rb (hunk 1 above)
Comment (tjwp) [inline spec/lib/refresh_spec.rb:97]: In this repo, there seems to be a preference to omit it. I tend to include it to be more explicit. Especially if it applies to more than one example.

Comment (hankwallace) [inline spec/lib/refresh_spec.rb:97]: I wasn't aware there was a default, so I had to look it up.

**Comment by hankwallace** [comment]
g2g

File: spec/lib/refresh_spec.rb [hunk 2]
```diff
@@ -26,57 +26,93 @@
...
-  context 'when the request returns an error' do
-    it 'raises an error' do
-      connection.stub(:post).and_return(failure_response)
+    context 'when host option is specified' do
+      before do
+        args.merge!(:host => 'test.salesforce.com')
+      end
 
-      expect { trigger }.to raise_error('Unauthorized: Access Denied')
+      it 'use overrides Metaforce.configuration.host' do
```
Comment (stevepescatore) [inline spec/lib/refresh_spec.rb]: This reads oddly. Maybe just "overrides Metaforce.configuration.host"?

Comment (tjwp) [inline spec/lib/refresh_spec.rb]: Good call.

**Comment by stevepescatore** [comment]
Looks good.

**Comment by tjwp** [comment]
Force pushed squashed commit.

**Pull request merged**